### PR TITLE
Change toDecimal return type

### PR DESCRIPTION
**Pull request by mmaurello**
### Description

Change return type of `toDecimal` function to `Big` type from `big.js`.

### Checklist

- [x] If this requires a documentation change, I have created a PR in [moonbeam-docs](https://github.com/PureStake/moonbeam-docs) repository.
- [x] If this requires it, I have updated the Readme
- [x] If necessary, I have updated the examples
- [x] I have verified if I need to create/update unit tests
- [x] I have verified if I need to create/update acceptance tests
- [x] If necessary, I have run acceptance tests on this branch in CI

Docs PR: 
https://github.com/PureStake/moonbeam-docs/pull/561

### CODE REVIEW

File: packages/utils/src/numbers/decimals.ts [hunk 1]
```diff
@@ -5,17 +5,22 @@ export function toDecimal(
...
-  return result.toNumber();
+  return result;
 }
 
 export function toBigInt(amount: string | number, decimals: number): bigint {
   const multiplier = Big(10).pow(decimals);
   const result = Big(amount).mul(multiplier);
 
-  return BigInt(result.toFixed());
+  return BigInt(result.toFixed(0, 0));
```
Comment (mmaurello) [inline packages/utils/src/numbers/decimals.ts:20]: If the method receives a number with more decimals than allowed, it throws an uncontrolled error when trying to create a bigint from a number with decimals. By passing` toFixed(0,0)` we ensure that the number being converted doesn't have decimals, effectively truncating the decimals on the right. 

Another approach is using the `hasDecimalOverflow` function to throw a controlled error but I feel like the first option is simpler and will work for most cases. We can debate this though

Comment (ekenigs) [inline packages/utils/src/numbers/decimals.ts:20]:
```suggestion
  return BigInt(result.toFixed(0, Big.roundDown));
```

Comment (ekenigs) [reply packages/utils/src/numbers/decimals.ts:20]: I think `toFixed(0,0)` is fine and should work.

File: packages/utils/src/numbers/decimals.test.ts [hunk 2]
```diff
@@ -87,6 +115,32 @@ describe('utils - decimals', () => {
...
+
+  describe('hasDecimalOverflow', () => {
+    it('should return true if the value has more decimals that allowed', () => {
+      expect(hasDecimalOverflow('66712.0367386073069948646', 18)).toBe(true);
+      expect(hasDecimalOverflow('66712.0367386073069948645', 18)).toBe(true);
+      expect(hasDecimalOverflow('12.036073069948646', 12)).toBe(true);
+      expect(hasDecimalOverflow('12.0360730692110', 12)).toBe(true);
+      expect(hasDecimalOverflow('12.036073069', 12)).toBe(false);
+      expect(hasDecimalOverflow('12.036073069210', 12)).toBe(false);
+      expect(hasDecimalOverflow('66712.036738607306994864', 18)).toBe(false);
```
Comment (ekenigs) [inline packages/utils/src/numbers/decimals.test.ts:135]: Should these 2 cases go to the second `it`?

**Review by ekenigs — CHANGES_REQUESTED** [review]

File: packages/utils/src/numbers/decimals.test.ts (hunk 2 above)
Comment (mmaurello) [reply packages/utils/src/numbers/decimals.test.ts]: Yes actually :)
done

**Review by ekenigs — APPROVED** [review]

**Pull request merged**